docs/tutorial/writing-tests/test-case.rst

```diff
@@ -0,0 +1,1105 @@
+          "\nlink_type: " + .link_type + "\n"'
+
+This test adds a new dependency to our provider. We need to declare this in
+the correct spot else this will not work in a reproducible manner. Lets create
+a packaging meta-data unit.
+
+.. code-block:: none
+
+    id: extended_tutorial_dependencies
+    unit: packaging meta-data
+    os-id: debian
+    Depends:
+      jq
+
+If you now run the following command you will notice a validation error.
+
+.. code-block:: none
+
+
+    (checkbox_venv) > python3 manage.py validate
+    [...]
+    error: ../base/units/submission/packaging.pxu:3: field 'Depends', clashes with 1 other unit, look at: ../base/units/submission/packaging.pxu:1-3, units/extended_tutorial.pxu:1-4
+    Validation of provider tutorial has failed
+
+Opening the file that the validator complains about, you will notice that the
+jq dependency is already required by a base provider test. We can rely on the
+base provider, so we can safely remove this dependency from our provider.
+
+.. warning::
+   If you don't have ``jq`` installed on your machine, install it now else you
+   won't be able to follow the next steps. You can install it either via
+   ``sudo snap install jq`` or ``sudo apt install jq``
+
+Now that we have this new resource lets run it to see what the output is
+
+.. code-block:: none
+
+    (checkbox_venv) >  checkbox-cli run com.canonical.certification::network_iface_info
+    ===========================[ Running Selected Jobs ]============================
+    =========[ Running job 1 / 1. Estimated time left (at least): 0:00:00 ]=========
+    ----------------[ Fetches information of all network intefaces ]----------------
+    ID: com.canonical.certification::network_iface_info
+    Category: com.canonical.plainbox::uncategorised
+    ... 8< -------------------------------------------------------------------------
+    interface: lo
+    link_info_kind:
+    link_type: loopback
+
+    interface: enp2s0f0
+    link_info_kind:
+    link_type: ether
+
+    interface: enp5s0
+    link_info_kind:
+    link_type: ether
+
+    interface: wlan0
+    link_info_kind:
+    link_type: ether
+
+    interface: lxdbr0
+    link_info_kind: bridge
+    link_type: ether
+
+    interface: veth993f2cd0
+    link_info_kind: veth
+    link_type: ether
+
+    interface: tun0
+    link_info_kind: tun
+    link_type: none
+
+We now add a ``requires:`` constraint to our jobs so that, if no interface
+that could possibly connected to the internet is on the machine, we can
+skip them instead of failing.
+
+.. code-block:: none
+    :emphasize-lines: 4,5
+
+    id: network_available
+    flags: simple
+    _summary: Test that the internet is reachable
+    requires:
+      network_iface_info.link_type == "ether"
+    command:
+      ping -c 1 1.1.1.1
+
+If we now run the ``network_available`` test, Checkbox will automatically pull
+also ``network_iface_info``. Note that this only happens because both are in
```

Comment:
```suggestion
If we now run the ``network_available`` test, Checkbox will also automatically
pull ``network_iface_info``. Note that this only happens because both are in
```